src/box/lua/config/instance_config.lua

```diff
@@ -638,25 +640,64 @@ return schema.new('instance_config', schema.record({
         end,
     }),
     iproto = schema.record({
-        -- XXX: listen/advertise are specific: accept a string of
-        -- a particular format, a number (port), a table of a
-        -- particular format.
-        --
-        -- Only a string is accepted for now.
-        listen = schema.scalar({
-            type = 'string',
+        listen = schema.array({
+            items = schema.record({
+                uri = schema.scalar({
+                    type = 'string',
+                    validate = function(data, w)
+                        -- Substitute variables with placeholders to don't
+                        -- confuse the URI parser with the curly brackets.
+                        data = data:gsub('{{ *.- *}}', 'placeholder')
+                        local uri, err = urilib.parse(data)
+                        if uri == nil then
+                            w.error('Unable to parse an URI: %s', err)
+                        end
+                        if uri.params ~= nil then
+                            err = "URI parameters should be described in " ..
+                                  "the 'params' field, not as the part of URI"
+                            w.error(err)
+                        end
+                    end,
+                }),
+                params = schema.record({
+                    transport = schema.enum({
+                        'plain',
+                        'ssl',
+                    }),
+                    -- Mandatory server options for TLS.
+                    ssl_key_file = enterprise_edition(schema.scalar({
+                        type = 'string',
+                    })),
+                    ssl_cert_file = enterprise_edition(schema.scalar({
+                        type = 'string',
+                    })),
+                    -- Optional server options for TLS.
+                    ssl_ca_file = enterprise_edition(schema.scalar({
+                        type = 'string',
+                    })),
+                    ssl_ciphers = enterprise_edition(schema.scalar({
+                        type = 'string',
+                    })),
+                    ssl_password = enterprise_edition(schema.scalar({
+                        type = 'string',
+                    })),
+                    ssl_password_file = enterprise_edition(schema.scalar({
+                        type = 'string',
+                    })),
+                }),
+            }, {
+                validate = function(data, w)
+                    -- No items in iproto.listen at all -- OK.
+                    if data == nil or next(data) == nil then
+                        return
+                    end
+                    -- There is some data -- the prefix should be there.
```

Comment:
Typo: prefix.

BTW, should an empty record be an error too? Here we have an array of such records and if there is no need to set the URI, one can just don't add the item.

Reply:
Thanks, fixed.

Reply:
How about the second point?

Reply:
Sorry, I forgot about it. Fixed.